src/components/HourlyWeather/HourlyWeatherSingle.jsx

```diff
@@ -1,9 +1,21 @@
 import { Box, Card, Divider, Paper, Typography } from '@mui/material';
 import { useSelector } from 'react-redux';
 import { transformHourlyInfo } from '../../helpers/transformWeatherData';
+import { getWeatherInfo } from '../../redux/slices/weatherSlice';
+import { useState, useEffect } from 'react';
 
 const HourlyWeatherSingle = (singleHourDataRaw) => {
-    const weatherInfo = useSelector((state) => state.weatherInfoSlice.value);
+    const weatherInfo = useSelector((state) => getWeatherInfo(state));
+    const [iconBGColor, setIconBGColor] = useState();
+
+    useEffect(() => {
```

Comment:
This is duplicated code (found the same code in other component). Could you please extract it in a custom hook?